File: arrays/moveZeroes.js
// [1,0,3,2,0]


function moveZeroes(arr) {
  // keep count of index
  index = 0

  // iterate array and put in element at index for every non zero
  for (var i = 0; i < arr.length; i++) {
    var current = arr[i];

    if (current !== 0) {
      arr[i] = 0;
      arr[index] = current;
      index++;
    }
  }

  return arr;
}

var arr = [3];
console.log(moveZeroes(arr));

File: arrays/moveZeroes.go
func moveZeroes(nums []int) {
  index := 0
  for i := 0; i < len(nums); i++ {
    current := nums[i];
    if current != 0 {
      nums[i] = 0
      nums[index] = current
      index++
    }
  }
}